sklearn/mixture/_gaussian_mixture.py

```diff
@@ -715,10 +715,10 @@ def _initialize(self, X, resp):
         weights, means, covariances = _estimate_gaussian_parameters(
             X, resp, self.reg_covar, self.covariance_type
         )
-        weights /= n_samples
```

Comment:
I suggest that, as with the other comment, the weights normalization is done here. We expect normalized weights to be input in `weights_init`.
```suggestion
       self.weights_ /= self.weights_.sum()
```
As per the negative weights, I would not suggest normalizing by absolute values. But by moving the normalization step here (only applying it for init, but not re-normalizing what might be carefully normalized user weights), we largely avoid this problem, as the current inputs shouldn't really provide negative weights as far as I am aware.

However, PR #24812 making GMM initialization more robust (allowing users to pass in their own responsibilities matrix, or to use a custom `callable`, such as a pre-fitted K-means) will make this matter. So it's worth solving the issue here.